code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -885,14 +885,17 @@ Pass a positive integer as an argument to override a bot's default speed.
 /mob/living/simple_animal/bot/ui_act(action, params)
 	. = ..()
 	if(.)
-		return
+		return TRUE
+
 	if(!bot_core.allowed(usr) && !usr.has_unlimited_silicon_privilege)
 		to_chat(usr, span_warning("Access denied."))
-		return
+		return TRUE
```

Comment:
This is a failure condition, why are you returning true?
I notice this in the rest of ui_act too, you're also not returning true when something is actually done relating to the ui, which is what you're meant to be doing

Reply:
TRUE here is just saying subtypes shouldn't continue their ui_act, so in general it shouldn't return unless parent's ui_act did nothing, because I don't want it running for no reason if the only thing they've actually done was from parent.

Reply:
Why should subtypes not continue their act?
Their may be a point where they want to, but can't because they can't walk back this true call, right? that feels really unintuitive to deal with

Reply:
The only reason I see it needed is if both parent and a subtype is changing the same thing, but I don't think that should be a thing either since theyre all buttons for turning a specific thing on and off to give/remove some flag or turn some var on/off.

Reply:
it's just like
what if someone wants to make a ui_act that is ok with some of the conditions you're outlawing here? they'll run into un-intuitive behavior that requires digging through parent ui act calls. I don't like that.

returning TRUE doesn't exist to let parent uis block children for this sort of thing, it exists so a ui_act can be marked as "handled", and trigger a rerender

This seems backwards to that (poorly communicated) goal, and I don't like what applying this pattern more generally would do

Reply:
I don't understand why a bot would be able to have its ui accessed by anyone despite the cover being locked, it's the point of the bot's cover being there. at worst, we can maybe just add the ability to permanently keep covers locked open.

Reply:
I guess an cover agnostic ui would be very different from what you have here...